File: 20A40/cir-agus/EJER_24/Referencia.java
class Referencia{
    int x = 0;

    Referencia incremento(){
        x++;
        return this;
    }
    
    public void meta(){
        System.out.print("\nEscribir y copilar el codigo de ");
        System.out.println("la presentacion 2.3");
        System.out.println("------------------------------");
    }

    public Referencia[] calculos(){

        Referencia r = new Referencia();
        Referencia alias;
        alias = r.incremento().incremento().incremento();
        alias.incremento().incremento();
        
        Referencia res[] = {r,alias};
        return res; 
    }

    public void salida(Referencia[] answ){
        System.out.println("r = "+answ[0].x);
        System.out.println("alias = "+answ[1].x);
    }

    public static void main(String args[]){
        Referencia cRefe =  new Referencia();

        cRefe.meta();
        cRefe.salida(cRefe.calculos());
    }
}

File: 20A40/cir-agus/EJER_26/Referencia.cpp
# include <iostream>
# include <tuple>

using namespace std;

class Referencia
{
    public:
    int x = 0;
    
    Referencia &incremento()
    {
        x++;
        return *this;
    }

};

void meta()
{
    cout<<"Hacer lo mismo que el 24 pero ahora en C++"<<endl;
}

tuple<Referencia,Referencia> calculos()
{
    
    Referencia r =  Referencia();
    Referencia alias;

    alias = r.incremento().incremento().incremento();
    alias.incremento().incremento();

    return make_tuple(r,alias);
}

void salida(Referencia rR, Referencia rA)
{
    cout <<"r= " << rR.x;
    cout<<endl<<"alias= " <<rA.x;
}

int main(int argc, char const *argv[])
{
    meta();
    tuple<Referencia,Referencia> myR = calculos();
    salida(get<0>(myR),get<1>(myR));

    return 0;
}
